src/main/java/se/michaelthelin/spotify/SpotifyHttpManager.java

```diff
@@ -402,6 +396,26 @@ public Builder setSocketTimeout(Integer socketTimeout) {
       return this;
     }
 
+    public Builder setConnectionManager(HttpClientConnectionManager connectionManager) {
+      this.connectionManager = connectionManager;
+      return this;
+    }
+
+    HttpClientConnectionManager getConnectionManager() {
+      if (connectionManager == null) {
+        BasicHttpClientConnectionManager basicHttpClientConnectionManager = new BasicHttpClientConnectionManager();
+        basicHttpClientConnectionManager.setConnectionConfig(ConnectionConfig
+          .custom()
+          .setConnectTimeout(this.connectTimeout != null ?
+            Timeout.ofMilliseconds(this.connectTimeout) :
+            ConnectionConfig.DEFAULT.getConnectTimeout())
+          .build());
+        this.connectionManager = basicHttpClientConnectionManager;
+      }
+
+      return connectionManager;
+    }
```

Comment:
I think this would return a new connection manager on each invocation. Wouldn't it make more sense to define a default connection manager once, as singleton, and then always return that instance if no custom connection manager is set?

Reply:
Ah yes, you're right, nvm!